sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -708,7 +720,11 @@ class GpuHashAggregateIterator(
      */
     def preProcess(toAggregateBatch: ColumnarBatch): ColumnarBatch = {
       withResource(new NvtxRange("pre-process", NvtxColor.DARK_GREEN)) { _ =>
-        GpuProjectExec.project(toAggregateBatch, preStepBound)
+        if (useTieredProject) {
```

Comment:
I wish, and it could be follow on work, that this could be handled by `GpuProjectExec.project`. E.g. that `preStepBoundTiered` could be a class, something like `TieredBoundReferences`,  that `.project` could be overloaded for. It would remove some of the changes here, and push the decision to use tiered to the binding part.

Reply:
I agree that would be better.  I may defer this to a follow-on PR though.